monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,320 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Optional
+
+import numpy as np
+
+from monai.config.type_definitions import DtypeLike, NdarrayOrTensor
+from monai.transforms.post.array import Activations, AsDiscrete, RemoveSmallObjects, SobelGradients
+from monai.transforms.transform import Transform
+from monai.transforms.utils_pytorch_numpy_unification import max, maximum, min
+from monai.utils import TransformBackends, convert_to_numpy, optional_import
+from monai.utils.type_conversion import convert_to_dst_type
+
+label, _ = optional_import("scipy.ndimage.measurements", name="label")
+disk, _ = optional_import("skimage.morphology", name="disk")
+opening, _ = optional_import("skimage.morphology", name="opening")
+watershed, _ = optional_import("skimage.segmentation", name="watershed")
+
+__all__ = ["Watershed", "GenerateMask", "GenerateProbabilityMap", "GenerateDistanceMap", "GenerateMarkers"]
+
+
+class Watershed(Transform):
+    """
+    Use `skimage.segmentation.watershed` to get instance segmentation results from images.
+    See: https://scikit-image.org/docs/stable/api/skimage.segmentation.html#skimage.segmentation.watershed.
+
+    Args:
+        connectivity: An array with the same number of dimensions as image whose non-zero elements indicate
+            neighbors for connection. Following the scipy convention, default is a one-connected array of
+            the dimension of the image.
+        dtype: target data content type to convert, default is np.uint8.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(self, connectivity: Optional[int] = 1, dtype: DtypeLike = np.uint8) -> None:
+        self.connectivity = connectivity
+        self.dtype = dtype
+
+    def __call__(  # type: ignore
+        self, image: NdarrayOrTensor, mask: Optional[NdarrayOrTensor] = None, markers: Optional[NdarrayOrTensor] = None
+    ) -> NdarrayOrTensor:
+        """
+        Args:
+            image: image where the lowest value points are labeled first. Shape must be [1, H, W, [D]].
+            mask: optional, the same shape as image. Only points at which mask == True will be labeled.
+                If None (no mask given), it is a volume of all 1s.
+            markers: optional, the same shape as image. The desired number of markers, or an array marking
+                the basins with the values to be assigned in the label matrix. Zero means not a marker.
+                If None (no markers given), the local minima of the image are used as markers.
+        """
+
+        image = convert_to_numpy(image)
+        markers = convert_to_numpy(markers)
+        mask = convert_to_numpy(mask)
+
+        instance_seg = watershed(image, markers=markers, mask=mask, connectivity=self.connectivity)
+
+        return convert_to_dst_type(instance_seg, image, dtype=self.dtype)[0]
+
+
+class GenerateMask(Transform):
```

Comment:
Based on the docstring this mask is only be used for watershed, so it might be better to name it `GenerateWatershedMask`.